torch/distributed/elastic/agent/server/api.py

```diff
@@ -930,7 +938,11 @@ def _exit_barrier(self):
         except SignalException as e:
             log.warning(f"Got termination signal: {e.sigval}")
             raise
-        except Exception:
-            log.exception(
+        except Exception as ex:
+            message = (
                 f"Error waiting on exit barrier. Elapsed: {time.time() - start} seconds"
             )
+            if "timeout" in str(ex).lower():
```

Comment:
Do we need to do this string parsing (versus catching a particular exception type) because PTD (specifically the c10d store) raises a `RuntimeError` rather than a `TimeoutError` when block-waiting on a key?

Reply:
@kiukchung  - Yeah, I was trying to avoid this but could not find a generic way to distinguish between timeouts vs. other errors in the store. I can suppress `log.exception` entirely since this happens during shutdown anyway. What do you think?

Reply:
hmm, but then we would lose the original exception stack trace no?

I guess you don't want something like this either:

```
log.warning(message, exc_info=ex)
```
or 
```
log.error(message, exc_info=ex)
```

since both of them would still cause the exception stack trace to appear in the logs

Reply:
Got it. One last question: We are not re-raising the exception here so doesn't the error handler (both on mast as well as that error summary table that gets printed out at the end) contain the correct root error?

Reply:
I see thanks for the explanation! I've approved the PR.

Reply:
Thank you @kiukchung!

Reply:
> I see thanks for the explanation! I've approved the PR.

@kiukchung - would you be able to help this get merged? Thank you!

Reply:
I approved the PR, are you not able to merge it?

Reply:
@kiukchung , no sorry, it says I do not have permissions to merge.

Reply:
To merge in PyTorch repo, you just leave a comment "@pytorchbot merge"
Look at closed PRs as an example.

But now there is a merge conflict, it needs to be resolved first.